apollo-api/src/main/java/com/spotify/apollo/RequestContext.java

```diff
@@ -65,10 +67,35 @@
    * </pre>
    *
    * @see System#nanoTime()
+   * @deprecated - prefer using the {@link #metadata()} to get this information. This method will be
+   *               removed in the next major release.
    */
+  @Deprecated
   default long arrivalTimeNanos() {
     // This is not a good default for real implementations. It is simply a catch-all
     // default to not break existing implementations.
     return System.nanoTime();
   }
+
+  /**
+   * Returns the metadata available for this request.
+   */
+  default RequestMetadata metadata() {
```

Comment:
Since this technically can break builds, maybe we should bump the minor version and mention it in the release note?